server/chat-commands.js

```diff
@@ -3796,6 +3796,42 @@ const commands = {
 	},
 	importinputloghelp: [`/importinputlog [inputlog] - Starts a battle with a given inputlog. Requires: + % @ & ~`],
 
+	acceptdraw: 'acceptdraw',
+	offerdraw(target, room, user, connection, cmd) {
+		const battle = room.battle;
+		if (!battle) return this.errorReply("Must be in a battle room.");
+		if (!Config.allowrequestingties) {
+			return this.errorReply("This server does not allow offering draws.");
+		}
+		if (!this.can('roomvoice', null, room)) return;
+		if (cmd === 'acceptdraw' && !battle.allowTie) return this.errorReply("No draw offer available. It might have been automatically withdrawn as the current turn started.");
+		const playerIds = Object.keys(battle.playerTable);
+		if (!battle.allowTie) {
+			battle.allowTie = [];
+			for (const player of playerIds) {
+				if (player === user.userid) {
+					battle.allowTie.push(player);
+					continue;
+				}
+				Users(player).sendTo(
+					room,
+					Chat.html`|html|${user.name} wants this game to be a tie; <button class="button" name="send" value="/acceptdraw">accept tie</button>?`
```

Comment:
I think this could be phrased better. What about something like:
"Your opponent wants this battle to end in a tie. If you want, you can ``<button>``accept this battle to end in a tie``</button>``"

Reply:
@panpawn I think your phrasing is pretty bad English, and also I don't like messages that are unnecessarily verbose. The shorter the message, the easier it is to read, and the less likely someone will skim over something important.

Reply:
At the least, I think this line should be clear that this will end the battle, and should use a word like "end"

Reply:
Fair, I wouldn't mind replacing "wants this game to be a tie" with "wants this game to end in a tie". I just think the "If you want" part is unnecessarily long.